pkg/labels/array.go

```diff
@@ -129,3 +129,30 @@ func (ls LabelArray) Get(key string) string {
 	}
 	return ""
 }
+
+// DeepCopy returns a deep copy of the labels.
+func (ls LabelArray) DeepCopy() LabelArray {
+	o := make(LabelArray, 0, len(ls))
+	for _, v := range ls {
+		if v == nil {
```

Comment:
It already is? in the sense that it won't crash with a nil. Did you mean DeepCopy on a nil would return a nil?
If you're wondering why I didn't use the Labels type since it has a DeepCopy already, it 's because there is no function to turn a LabelArray into a Labels but the Rule objects have LabelArrays (so, I figured writing the copy would be closer to the what we had before than making a converter).

Reply:
> It already is? in the sense that it won't crash with a nil. Did you mean DeepCopy on a nil would return a nil?

Yes, if you return nil on nil you don't need to embed the nil conditional everywhere

Reply:
done